compiler/rustc_typeck/src/collect.rs

```diff
@@ -1767,7 +1767,7 @@ fn fn_sig(tcx: TyCtxt<'_>, def_id: DefId) -> ty::PolyFnSig<'_> {
                     }
                     diag.emit();
 
-                    ty::Binder::bind(fn_sig, tcx)
+                    ty::Binder::dummy(fn_sig)
```

Comment:
Can you just move this up to right under `fn_sig` is created? Then when used, just `skip_binder()`